lib/vector-core/src/event/metric/tags.rs

```diff
@@ -71,24 +70,28 @@ impl TagValueSet {
     }
 
     #[cfg(test)]
-    fn contains(&self, value: &str) -> bool {
+    fn contains<T>(&self, value: &T) -> bool
+    where
+        T: Eq + Hash + PartialEq<Option<String>>,
```

Comment:
Would it make sense to use the `TagValue` type alias here? At least just for consistency.